gradle/quality.gradle

```diff
@@ -0,0 +1,42 @@
+//apply plugin: 'findbugs'
+//apply plugin: 'checkstyle'
+//apply plugin: 'pmd'
+//
+//tasks.withType(FindBugs) {
+//    excludeFilter = file("$rootProject.projectDir/config/findbugs/excludeFilter.xml")
+//}
+//
+//// ignore test source for checkstyles and pmd
+//checkstyle.sourceSets = [sourceSets.main]
+//pmd.sourceSets = [sourceSets.main]
+//
+//pmd {
+//    ruleSets = [
+//            'java-basic',
+//            'java-braces',
+//            'java-clone',
+////            'java-codesize',
+////	            'java-comments',
+////	            'java-controversial',
+////	            'java-coupling',
+////	            'java-design',
+//            'java-empty',
+//            'java-finalizers',
+//            'java-imports',
+////	            'java-j2ee',
+////	            'java-javabeans',
+////	            'java-junit',
+////	            'java-logging-jakarta-commons',
+////	            'java-logging-java',
+//            'java-migrating',
+////	            'java-naming',
+////	            'java-optimizations',
+////	            'java-strictexception',
+//            'java-strings',
+////	            'java-sunsecure',     // complexity issues to be address + encapsulating byte arrays
```

Comment:
Nit: Indents look crazy. Would it be easier to read if we split the active and commented out rules into two separately alphabetized blocks?